crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -120,9 +141,74 @@ where
             is_valid
         })
     }
+
+    async fn update_reorged_logs(&self, pubsub: EthPubSub<Provider, Pool, Events, Network>) {
+        let mut stream = self.reorged_logs_stream(pubsub).await;
+        while let Some((id, logs)) = stream.next().await {
+            let mut filters = executor::block_on(self.active_filters().inner.lock());
+            let active_filter =
+                filters.get_mut(&id).ok_or(FilterError::FilterNotFound(id)).unwrap();
+            if let FilterKind::Log(ref mut filter) = active_filter.kind {
+                filter.reorged_logs = Arc::new(Mutex::new(Some(logs)));
+            }
+        }
+    }
+
+    /// Reacts to reorged blocks, checks impacted log filters, stores reorged logs in the log filter
+    pub async fn reorged_logs_stream(
+        &self,
+        pubsub: EthPubSub<Provider, Pool, Events, Network>,
```

Comment:
I'm not exactly sure what this does.

this looks a bit convoluted.

how the on reorg logic should work is:
1. listen for reorged blocks:
2. acquire a lock on the shared state
3. try to find matching filters, if so put the logs into the ActiveFilter

Reply:
The temp_reorged_blocks stores consecutive reorged blocks as they come. The matching logs for the given filter are stored in reverted_logs. If the hash of one of these blocks matches with the last polled hash of the filter, we know that the filter previously served all the blocks stored in temp_reorged_blocks. We can push the reverted logs in the reorged_logs vector along with the filter id. If the last polled hash does not match before the block subscription receives a newly included block, we can simply leave it like that and clear the temp_reorged_blocks. The  filter did not serve reorged blocks.